.github/workflows/ci-core.yml

```diff
@@ -244,6 +244,93 @@ jobs:
+        uses: actions/setup-node@v4.0.4
+      - name: Setup NodeJS
+        if: ${{ needs.filter.outputs.changes == 'true' }}
+        uses: ./.github/actions/setup-nodejs
+        with:
+          prod: "true"
+      - name: Setup Go
+        if: ${{ needs.filter.outputs.changes == 'true' }}
+        uses: ./.github/actions/setup-go
+      - name: Setup Postgres
+        if: ${{ needs.filter.outputs.changes == 'true' }}
+        uses: ./.github/actions/setup-postgres
+      - name: Touching core/web/assets/index.html
+        if: ${{ needs.filter.outputs.changes == 'true' }}
+        run: mkdir -p core/web/assets && touch core/web/assets/index.html
+      - name: Download Go vendor packages
+        if: ${{ needs.filter.outputs.changes == 'true' }}
+        run: go mod download
+      - name: Replace chainlink-evm deps
+        if: ${{ needs.filter.outputs.changes == 'true' && inputs.evm-ref != ''}}
+        shell: bash
+        run: go get github.com/smartcontractkit/chainlink-integrations/evm/relayer@${{ inputs.evm-ref }}
+      - name: Build binary
+        if: ${{ needs.filter.outputs.changes == 'true' }}
+        run: go build -o chainlink.test .
+      - name: Setup DB
+        if: ${{ needs.filter.outputs.changes == 'true' }}
+        run: ./chainlink.test local db preparetest
+      - name: Load test outputs
+        if: ${{ needs.filter.outputs.changes == 'true' }}
+        uses: actions/download-artifact@v4.1.8
+        with:
+          name: go_core_tests_logs
+          path: ./artifacts
+      - name: Delete go_core_tests_logs/coverage.txt
+        if: ${{ needs.filter.outputs.changes == 'true' }}
+        shell: bash
+        run: |
+          # Need to delete coverage.txt so the disk doesn't fill up
+          rm -f ./artifacts/go_core_tests_logs/coverage.txt
+      - name: Build flakey test runner
+        if: ${{ needs.filter.outputs.changes == 'true' }}
+        run: go build ./tools/flakeytests/cmd/runner
+      - name: Re-run tests
+        if: ${{ needs.filter.outputs.changes == 'true' }}
+        env:
+          GRAFANA_INTERNAL_BASIC_AUTH: ${{ secrets.GRAFANA_INTERNAL_BASIC_AUTH }}
+          GRAFANA_INTERNAL_HOST: ${{ secrets.GRAFANA_INTERNAL_HOST }}
+          GITHUB_EVENT_PATH: ${{ github.event_path }}
+          GITHUB_EVENT_NAME: ${{ github.event_name }}
+          GITHUB_REPO: ${{ github.repository }}
+          GITHUB_RUN_ID: ${{ github.run_id }}
+        run: |
+          ./runner \
+            -grafana_auth=$GRAFANA_INTERNAL_BASIC_AUTH \
+            -grafana_host=$GRAFANA_INTERNAL_HOST \
+            -gh_sha=$GITHUB_SHA \
+            -gh_event_path=$GITHUB_EVENT_PATH \
+            -gh_event_name=$GITHUB_EVENT_NAME \
+            -gh_run_id=$GITHUB_RUN_ID \
+            -gh_repo=$GITHUB_REPO \
+            -command=./tools/bin/go_core_tests \
+            `ls -R ./artifacts/output.txt`
+      - name: Store logs artifacts
+        if: ${{ needs.filter.outputs.changes == 'true' && always() }}
```

Comment:
curious: how does that actually work? some condition + `always()`?

Reply:
i think it means this step will run regardless if the previous steps failed or not provided the `changes==true`